llmfoundry/utils/builders.py

```diff
@@ -76,8 +76,11 @@ def build_icl_data_and_gauntlet(
     return icl_evaluators, logger_keys, eval_gauntlet_cb
 
 
-def build_callback(name: str, kwargs: Union[DictConfig, Dict[str,
-                                                             Any]]) -> Callback:
+def build_callback(
+    name: str,
+    kwargs: Union[DictConfig, Dict[str, Any]],
+    config: Dict[str, Any],
```

Comment:
should we default `config` to `None` since only some callbacks (i.e. yours) needs it? and raise error if not provided when making async eval? fine either way, i think this api is not critical to keep backwards compat on

Reply:
Yea it didn't feel too important to keep backwards compat since it is only used for training. Tests are already updated, would prefer to keep as is